hooks/use-dappnode-deposit.ts

```diff
@@ -0,0 +1,237 @@
+  const [hasDuplicates, setHasDuplicates] = useState(false);
+  const [isBatch, setIsBatch] = useState(false);
+  const [filename, setFilename] = useState("");
+  const client = getPublicClient(config, { chainId: chainId as 100 });
+
+  const { data: user }: { data: DappnodeUser | undefined } = useReadContract({
+    abi: dappnodeIncentiveABI,
+    address: contractConfig?.addresses.dappnodeIncentive,
+    functionName: "users",
+    args: [address],
+  });
+
+  const isWrongNetwork = chainId !== 100;
+  const { data: depositHash, writeContractAsync, isPending, isError } = useWriteContract();
+  const { isSuccess: depositSuccess } = useWaitForTransactionReceipt({
+    hash: depositHash,
+  });
+
+  const dappnodeValidate = useCallback(
+    async (deposits: DepositDataJson[]) => {
+      let newDeposits = [];
+      let hasDuplicates = false;
+      let _isBatch = false;
+      if (contractConfig && user) {
+        const checkJsonStructure = (depositDataJson: DepositDataJson) => {
+          return (
+            depositDataJson.pubkey &&
+            depositDataJson.withdrawal_credentials &&
+            depositDataJson.amount &&
+            depositDataJson.signature &&
+            depositDataJson.deposit_message_root &&
+            depositDataJson.deposit_data_root &&
+            depositDataJson.fork_version
+          );
+        };
+
+        if (!deposits.every) {
+          throw Error(
+            "Oops, something went wrong while parsing your json file. Please check the file and try again."
+          );
+        }
+
+        if (
+          deposits.length === 0 ||
+          !deposits.every((d) => checkJsonStructure(d))
+        ) {
+          throw Error("This is not a valid file. Please try again.");
+        }
+
+        if (
+          !deposits.every((d) => d.fork_version === contractConfig.forkVersion)
+        ) {
+          throw Error(
+            "This JSON file isn't for the right network (" +
+              deposits[0].fork_version +
+              "). Upload a file generated for you current network: " +
+              chainId
+          );
+        }
+                
+        if (!deposits.every((d) => `0x`+d.withdrawal_credentials.substring(24).toLocaleLowerCase() === user[0].toLowerCase())) {
+          throw Error(
+            "Atleast one of the provided keys does not match your safe address as withdrawal credentials."
+          );
+        }
+        if (deposits.length !== user[2]) {
+          throw Error(
+            `Wrong number of keys. Expected claiming (${user[2]}) validator deposits to your safe.`
+          );
+        }
+
+        const { deposits: existingDeposits, lastBlock: fromBlock } =
+          await loadCachedDeposits(
+            chainId,
+            contractConfig.depositStartBlockNumber
+          );
+
+        const events = await fetchDeposit(
+          contractConfig.addresses.deposit,
+          fromBlock,
+          client
+        );
+
```

Comment:
Also not connected to this PR, as subgraph wasn't implemented yet, but when indexer will be added, this part will be changed too.